src/asmjit/core/virtmem.cpp

```diff
@@ -339,7 +371,12 @@ class AnonymousMemory {
     static volatile uint32_t memfd_create_not_supported;
 
     if (!memfd_create_not_supported) {
-      _fd = (int)syscall(__NR_memfd_create, "vmem", 0);
```

Comment:
I would like to keep the syscall here.

I think if it's replaced by memfd_create() somebody would be unhappy as this syscal was first wrapped by glibc 2.27. So I think it's just safer to keep the syscall as then it's not that important which version glibc has.

Reply:
Sure, I can restore that.  Does the desired end-state also include removing the includes of `linux/memfd.h` and `linux/mman.h`?  If so, I'd just add some constant definitions, not unlike what Perfetto did here

https://github.com/google/perfetto/blob/master/src/tracing/ipc/memfd.h#L27-L31

in place of those includes so we can request the relevant HugeTLB options, `MFD_HUGETLB` and `MFD_HUGE_2MB`, symbolically.

Reply:
Yeah no problem with defining these constants.

What I would like is to keep the current compatibility if it's possible, so no raising glibc version, etc... if it's possible and if it doesn't add so much burden.

Reply:
Great.  I wasn't aware of the more limited glibc compatibility.  Anyway, it is not much of a burden at all for this to be standalone.  I'll go ahead and make the change.

Reply:
I use a modern Linux distro, so I almost never see these problems - but CI, weird Linux images used by some companies, etc... it's pretty wild world :)